Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/dialogController/MainController.java

```diff
@@ -149,37 +152,46 @@ public MainController() {
 
         portalSession = new PortalSession(this);
 
-        final String urlDecoded = Executions.getCurrent().getParameter("encodedToken");
-        LOGGER.info("\n\nIn PORTAL MainController, urlDecoded {}", urlDecoded);
-
         String usernameParsed = null;
 
         if (usingKeycloak) {
-            final String appAuthHeader = SecuritySsoHelper.getAppAuthHeader();
-            final String signedAppAuthHeader = SecuritySsoHelper.getSignedAppAuthHeader();
-
-            MainController.encKey = SecuritySsoHelper.getEnvEncKey();
-
             try {
-                if (StringUtils.isNotBlank(urlDecoded)) {
-                    usernameParsed = SecuritySsoHelper.getSsoUsername(urlDecoded, encKey);
+                final HttpServletRequest httpServletRequest =
+                        (HttpServletRequest) Executions.getCurrent().getNativeRequest();
+                final String appAuthHeader =
+                        JwtHelper.readCookie(httpServletRequest, "App_Auth");
+                LOGGER.info("\nIn MainController Read App_Auth cookie: {}", appAuthHeader);
+
+                final JWTClaimsSet jwtClaimsSet = JwtHelper.getClaimsSetFromJWT(appAuthHeader);
+                final String issuedAtStr = (String)jwtClaimsSet.getStringClaim(
+                        JwtHelper.STR_JWT_KEY_ISSUED_AT);
+                LOGGER.info("issuedAtStr {}", issuedAtStr);
+                final String expiryAtStr = (String)jwtClaimsSet.getStringClaim(
+                        JwtHelper.STR_JWT_EXPIRY_TIME);
+                LOGGER.info("expiryAtStr {}", expiryAtStr);
+
+                final boolean jwtHasExpired = JwtHelper.isJwtExpired(jwtClaimsSet, issuedAtStr, expiryAtStr);
+
+                if (jwtHasExpired) {
+                    LOGGER.info("In MainController, JWT IS expired");
```

Comment:
You don't really need to specify "In MainController" as part of the logging message; the LOGGER was given this information when it was created by the factory.

Reply:
Done.